pkg/store/consul/kv.go

```diff
@@ -232,6 +232,34 @@ func (c consulStore) SetPodTxn(ctx context.Context, podPrefix PodPrefix, nodenam
 	})
 }
 
+func (c consulStore) CASPod(podPrefix PodPrefix, nodename types.NodeName, manifest manifest.Manifest, modifyIndex uint64) error {
+	buf := bytes.Buffer{}
+	err := manifest.Write(&buf)
+	if err != nil {
+		return err
+	}
+
+	key, err := podPath(podPrefix, nodename, manifest.ID())
+	if err != nil {
+		return err
+	}
+	keyPair := &api.KVPair{
+		Key:         key,
+		Value:       buf.Bytes(),
+		ModifyIndex: modifyIndex,
+	}
+
+	ok, _, err := c.client.KV().CAS(keyPair, nil)
```

Comment:
I think you're going to not only need to check the index is 0 but also lock it with a session (so that the key gets rolled back if the session dies). It might be better to use a transaction here with the "lock" operation as well as a "check-index" operation

Reply:
I had the session creation and locking in the code that will call this function. Would that go better here? I'm a little unclear on the session stuff